### PR TITLE
Implement HelpRequestCommand (Resolves #2034)

### LINKED ISSUE #2034: Dodanie komendy /PP  -  kordynaty członka gildii z  !! czy tam @
Witam mozna by było dodac na bossbar i actionbar   kordynaty gracza np gracz {nick} potrzebuje pomocy {x} {y} {z}  
i wtedy w cfg by sie ustawiało czy ma sie to wyswietlac na AC CZY BOSSB
i ten bossbar by np znikał po 15 sek i ten bossbar byl by jako timer bossbar w sensie ze pasek by szedl w dol 

 jesli ktos wpisze /pp to sie to wyswietla wszystkim członką g 
 jest to bardziej przydatne niz ta @ czy !! na chacie poniewaz jak jest duzo ludzi to czat zapierdziela i czasami trudno jest wylapac taka wiadomosc  i fajnie by było jak by było to  na actionbar lub bossbar

### PR DESCRIPTION
**Pull request by oskarscot**
Dear FunnyGuilds team,

I am thrilled to submit this pull request to suggest the addition of a new command to the FunnyGuilds project - the /helprequest (pp) command.

I believe that this new command will be game-changing for the project, as it will allow players to easily request help from other members of their guild. With the /helprequest command, players will be able to send a notification to their guild members, informing them of their need for assistance. This will make it easier for players to get help when they need it, and will promote a stronger sense of community within the guild.

Additionally, the /helprequest command will help to improve the overall gameplay experience by reducing the amount of time players spend waiting for assistance. Rather than having to rely on chance encounters with other players or on lengthy discussions in chat, players can simply use the /helprequest command to get the help they need quickly and easily.

I am confident that the addition of the /helprequest (pp) command will be a significant improvement to the FunnyGuilds project, and I look forward to seeing it implemented in the near future.

Thank you for your time and consideration to add a PP into the FunnyGuilds project.

Sincerely,
oskarscot

### CODE REVIEW

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/FunnyCommandsConfiguration.java [hunk 1]
```diff
@@ -33,34 +33,7 @@
...
-import net.dzikoysk.funnyguilds.feature.command.user.PlayerInfoCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.PvPCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.RankResetCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.RankingCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.SetBaseCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.StatsResetCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.TntCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.TopCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.ValidityCommand;
-import net.dzikoysk.funnyguilds.feature.command.user.WarCommand;
```
Comment (oskarscot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/FunnyCommandsConfiguration.java:63]: intellij moment

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java [hunk 2]
```diff
@@ -843,6 +843,17 @@ public static class Move extends OkaeriConfig {
                                 .build())
                         .build();
 
+                @Comment("Dostępne zmienne: {PLAYER}, {X}, {Y}, {Z}")
+                public SendableMessage helpRequest = SendableMessage.builder()
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java]: przenieś to do sekcji `guild.commands`, też możesz dać na sam dół

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java [hunk 3]
```diff
@@ -843,6 +843,17 @@ public static class Move extends OkaeriConfig {
...
+                        .chat("&7Gracz &a{PLAYER} &7poprosił o pomoc&7! &aX: {X} Y: {Y} Z: {Z}")
+                        .addHolders(BossBarHolder.builder("&7Gracz &a{PLAYER} &7poprosił o pomoc&7! &aX: {X} Y: {Y} Z: {Z}")
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java]: daj defaultowo na actionbara

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    @FunnyCommand(
+            name = "${user.pp.name}",
+            description = "${user.pp.description}",
+            aliases = "${user.pp.aliases}",
+            permission = "funnyguilds.helprequest",
+            acceptsExceeded = true,
+            playerOnly = true
+    )
+    public void execute(Player player, @IsMember User member){
```
Comment (Rollczi) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java]: popraw styl kodu:
```suggestion
    public void execute(Player player, @IsMember User member) {
```

**Review by P3ridot — CHANGES_REQUESTED** [review]

**Review by Rollczi — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/FunnyCommandsConfiguration.java [hunk 5]
```diff
@@ -111,7 +112,8 @@ public static FunnyCommands createFunnyCommands(FunnyGuilds plugin) {
                 .command("top", commands.top, TopCommand.class)
                 .command("validity", commands.validity, ValidityCommand.class)
                 .command("war", commands.war, WarCommand.class)
-                .command("tnt", commands.tnt, TntCommand.class);
+                .command("tnt", commands.tnt, TntCommand.class)
+                .command("pp", commands.helpRequest, HelpRequestCommand.class);
```
Comment (Rollczi) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/FunnyCommandsConfiguration.java]: `pp` to tylko skrótowy alias, zmień na `helprequest` żeby było to zgodnie tak jak w innych komendach

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import net.dzikoysk.funnyguilds.shared.FunnyFormatter;
+import net.dzikoysk.funnyguilds.user.User;
+import org.bukkit.entity.Player;
+
+public class HelpRequestCommand extends AbstractFunnyCommand {
+
+    @FunnyCommand(
+            name = "${user.pp.name}",
+            description = "${user.pp.description}",
+            aliases = "${user.pp.aliases}",
```
Comment (Rollczi) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java]: tutaj tak samo z tym `pp`

**Review by Rollczi — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java [hunk 7]
```diff
@@ -842,11 +843,16 @@ public static class Move extends OkaeriConfig {
                                 .clearOtherBars(true)
                                 .build())
                         .build();
-
             }
 
         }
 
+        @Comment("Dostępne zmienne: {PLAYER}, {X}, {Y}, {Z}")
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java]: Dalej jest w złym miejscu

**Review by P3ridot — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java [hunk 8]
```diff
@@ -768,6 +764,12 @@ public static class DamageTracking extends OkaeriConfig {
     @CustomKey("damage-ally")
     public boolean damageAlly = false;
 
+    @NegativeOrZero
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java]: Raczej powinno być @PositiveOrZero

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java [hunk 9]
```diff
@@ -11,12 +11,8 @@
 import eu.okaeri.configs.annotation.Names;
 import eu.okaeri.configs.exception.OkaeriException;
 import eu.okaeri.configs.serdes.commons.duration.DurationSpec;
-import eu.okaeri.validator.annotation.DecimalMax;
-import eu.okaeri.validator.annotation.DecimalMin;
-import eu.okaeri.validator.annotation.Min;
-import eu.okaeri.validator.annotation.NotBlank;
-import eu.okaeri.validator.annotation.Positive;
-import eu.okaeri.validator.annotation.PositiveOrZero;
+import eu.okaeri.validator.annotation.*;
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java]: Nie chcemy wildcard importów

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java [hunk 10]
```diff
@@ -11,12 +11,8 @@
...
+
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java]: niepotrzebna spacja

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java [hunk 11]
```diff
@@ -842,7 +849,6 @@ public static class Move extends OkaeriConfig {
                                 .clearOtherBars(true)
                                 .build())
                         .build();
-
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/message/MessageConfiguration.java]: ta spacja po coś tu była 😠

**Review by P3ridot — CHANGES_REQUESTED** [review]

File: plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+            name = "${user.helprequest.name}",
+            description = "${user.helprequest.description}",
+            aliases = "${user.helprequest.aliases}",
+            permission = "funnyguilds.helprequest",
+            acceptsExceeded = true,
+            playerOnly = true
+    )
+    public void execute(Player player, @IsMember User member) {
+        if (this.requestCooldown.cooldown(player, this.config.helpRequestCooldown)) {
+            return;
```
Comment (Rollczi) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java:25]: zostanie zwrócona tutaj wiadomość?

Comment (oskarscot) [reply plugin/src/main/java/net/dzikoysk/funnyguilds/feature/command/user/HelpRequestCommand.java:25]: nie, pozdrawiam

**Comment by oskarscot** [comment]
Implements #2034 (didnt mention issue smh)

File: plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java [hunk 13]
```diff
@@ -768,6 +768,12 @@ public static class DamageTracking extends OkaeriConfig {
     @CustomKey("damage-ally")
     public boolean damageAlly = false;
 
+    @PositiveOrZero
+    @Comment("")
+    @Comment("Co ile można użyć komendy /helprequest")
+    @CustomKey("help-request-cooldown")
```
Comment (P3ridot) [inline plugin/src/main/java/net/dzikoysk/funnyguilds/config/PluginConfiguration.java]: Ten CustomKey jest zbędny

**Review by P3ridot — CHANGES_REQUESTED** [review]

**Review by P3ridot — APPROVED** [review]

**Review by Rollczi — APPROVED** [review]

**Pull request merged**